### PR TITLE
Remove `dynapath`, accomodate `enrich-classpath`

### PR DESCRIPTION
**Pull request by vemv**
Fixes https://github.com/clojure-emacs/orchard/issues/122
Fixes https://github.com/clojure-emacs/orchard/issues/120
Fixes https://github.com/clojure-emacs/orchard/issues/105
Fixes https://github.com/clojure-emacs/orchard/issues/103
Fixes https://github.com/clojure-emacs/orchard/issues/50

### CODE REVIEW

File: src/orchard/java.clj [hunk 1]
```diff
@@ -44,58 +42,27 @@
...
-                     (some-> (io/resource path)
-                             ^JarURLConnection (. openConnection)
-                             (. getJarFileURL)))]
-      (or (base-url "java.base/java/lang/Object.java") ; JDK9+
-          (base-url "java/lang/Object.java")           ; JDK8-
-          (jdk-find "src.zip")))))
+  [_]
+  nil)
+
+(def ^:deprecated jdk-sources nil)
```
Comment (vemv) [inline src/orchard/java.clj:53]: From what I pondered on previous iterations: ^:deprecated + no-op impls are the least breaking choice

(as opposed to an `(assert false)`)

File: src/orchard/java/classpath.clj [hunk 2]
```diff
@@ -56,26 +50,26 @@
        (.split (System/getProperty "java.class.path")
                (System/getProperty "path.separator"))))
 
+(defn classpath-urls [classloader]
```
Comment (vemv) [inline src/orchard/java/classpath.clj:53]: This is (essentially) what `dp/classpath-urls` was doing

File: test-newer-jdks/orchard/java/parser_test.clj [hunk 3]
```diff
@@ -1,66 +1,50 @@
 (ns orchard.java.parser-test
   (:require
+   [clojure.test :refer [deftest is testing]]
    [orchard.java.parser :as parser]
-   [clojure.test :refer [deftest is testing]]))
+   [orchard.test.util :as util])
+  (:import
+   (orchard.java DummyClass)))
 
-(defn compile-class-from-source
```
Comment (vemv) [inline test-newer-jdks/orchard/java/parser_test.clj:6]: Now compilation for this test case happens via Lein :java-source-paths, it reflects typical end-user usage and also exercises what enrich-classpath does / is able to do (it's not able to modify classpaths at runtime, and we don't want that either!)

**Review by bbatsov — APPROVED** [review]
Looks good  to me! Great work!

**Pull request merged**